docs/builtin-classes.md

```diff
@@ -0,0 +1,396 @@
+<!--- TEST_NAME BuiltinClassesTest -->
+
+# Builtin classes
+
+This is the second chapter of the [Kotlin Serialization Guide](serialization-guide.md).
+In addition to all the primitive types and strings, serialization for some classes from the Kotlin standard library, 
+including the standard collections, is built into the Kotlin Serialization. This chapter explains the details.
+
+**Table of contents**
+
+<!--- TOC -->
+
+* [Primitives](#primitives)
+  * [Numbers](#numbers)
+  * [Special floating-point values](#special-floating-point-values)
+  * [Long numbers](#long-numbers)
+  * [Long numbers as strings](#long-numbers-as-strings)
+  * [Enum classes](#enum-classes)
+  * [Serial names of enum entries](#serial-names-of-enum-entries)
+* [Composites](#composites)
+  * [Pair and triple](#pair-and-triple)
+  * [Lists](#lists)
+  * [Sets and other collections](#sets-and-other-collections)
+  * [Deserializing collections](#deserializing-collections)
+  * [Maps](#maps)
+
```

Comment:
Maybe add a short discussion of Object and Unit serialization. In particular noting that objects are serialized as empty structures (with a distinct type so formats can handle it as they see fit), but the state of objects is not serialized..

Reply:
Good point!